### ISSUE TITLE
prepareQuery got the error: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by oaktreecode** [ent/ent#3337]
I got issue when updated to the latest version

```
// Edges of the SysCategory.
func (SysCategory) Edges() []ent.Edge {
	return []ent.Edge{
		edge.To("children", SysCategory.Type).
			From("parent").
			Field("parent_id").
			Unique(),
}
```

and then I query

```
query category {
  sysCategories(first:1){
    edges {
      node {
        id
       name
 
        
        parent {
          id
        }
       
        parentID
        
      }
      }
    }
  
}
```

```
go 1.18

require (
	entgo.io/contrib v0.3.4
	entgo.io/ent v0.11.8
	github.com/99designs/gqlgen v0.17.24
	github.com/casbin/casbin/v2 v2.63.0
	github.com/gin-gonic/gin v1.8.2
	github.com/go-kratos/kratos/v2 v2.5.4
	github.com/go-resty/resty/v2 v2.7.0
	github.com/go-sql-driver/mysql v1.7.0
	github.com/google/uuid v1.3.0
	github.com/google/wire v0.5.0
	github.com/hashicorp/go-multierror v1.1.1
	github.com/natefinch/lumberjack v2.0.0+incompatible
	github.com/pkg/errors v0.9.1
	github.com/vektah/gqlparser/v2 v2.5.1
	github.com/vmihailenco/msgpack/v5 v5.3.5
	go.uber.org/automaxprocs v1.5.1
	go.uber.org/zap v1.24.0
	golang.org/x/exp v0.0.0-20230213192124-5e25df0256eb
	google.golang.org/genproto v0.0.0-20230209215440-0dfe4f8abfcc
	google.golang.org/grpc v1.53.0
	google.golang.org/protobuf v1.28.1
)
```

got the error from :
**GolandProjects/monitor-sys/internal/data/ent/syscategory_query.go:575**

```

 func (scq *SysCategoryQuery) prepareQuery(ctx context.Context) error {
	for _, inter := range scq.inters {
		if inter == nil {
			return fmt.Errorf("ent: uninitialized interceptor (forgotten import ent/runtime?)")
		}
		if trv, ok := inter.(Traverser); ok {
			if err := trv.Traverse(ctx, scq); err != nil {
				return err
			}
		}
	}
	for _, f := range scq.ctx.Fields { //----> line: 575
		if !syscategory.ValidColumn(f) {
			return &ValidationError{Name: f, err: fmt.Errorf("ent: invalid field %q for query", f)}
		}
	}
	if scq.path != nil {
		prev, err := scq.path(ctx)
		if err != nil {
			return err
		}
		scq.sql = prev
	}
	return nil
}

```
 

2023/02/21 23:41:02 driver.Query: query=SELECT `sys_categories`.`id`, `sys_categories`.`created_at`, `sys_categories`.`updated_at`, `sys_categories`.`deleted_at`, `sys_categories`.`name`, `sys_categories`.`code`, `sys_categories`.`weight`, `sys_categories`.`status`, `sys_categories`.`parent_id`, `sys_categories`.`icon_uri`, `sys_categories`.`body`, `sys_categories`.`extra_data` FROM `sys_categories` ORDER BY `sys_categories`.`id` ASC LIMIT 2 args=[]
runtime error: invalid memory address or nil pointer dereference

goroutine 15 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x65
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x19
github.com/99designs/gqlgen/graphql.DefaultRecover({0x10000e8ad?, 0x101103600?}, {0x10118bec0?, 0x102047f50?})
        /Users/x/go/pkg/mod/github.com/99designs/gqlgen@v0.17.24/graphql/recovery.go:17 +0x71
github.com/99designs/gqlgen/graphql.(*OperationContext).Recover(0xc0000acae0?, {0x1016406b0, 0xc000b32690}, {0x10118bec0?, 0x102047f50?})
        /Users/x/go/pkg/mod/github.com/99designs/gqlgen@v0.17.24/graphql/context_operation.go:124 +0x3e
 /api/graphql/generated.(*executionContext)._Query_sysCategories.func1()
        /Users/x/GolandProjects/x-sys/api/graphql/generated/generated.go:13649 +0x6a
panic({0x10118bec0, 0x102047f50})
        /usr/local/go/src/runtime/panic.go:884 +0x213
 /internal/data/ent.(*SysCategoryQuery).prepareQuery(0xc000384dc0, {0x1016406b0, 0xc000b32720})
       GolandProjects/x-sys/internal/data/ent/syscategory_query.go:575 +0x125
 /internal/data/ent.(*SysCategoryQuery).All(0xc000384dc0, {0x1016406b0?, 0xc000b32720?})

- [ ] The issue is present in the latest release.
- [ ] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Steps to Reproduce 🕹

Steps:

1.
2.
3.
4.

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.17.?  |
| Ent         | 0.9.?   |
| Database    | MySQL   |
| Driver      | https://github.com/go-sql-driver/mysql        |

**Comment by giautm** [comment]
Hello @oaktreecode, this issue was fixed by https://github.com/ent/contrib/pull/428 and released in the latest version ([v0.3.5](https://github.com/ent/contrib/compare/v0.3.4...v0.3.5)). Can you please upgrade `ent/contrib` and let me know if you still facing the problem?

**Comment by giautm** [comment]
Close this now, if you still have problem. Please let me known, I happy to help. Thank so much!
